plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -750,18 +845,30 @@ fu_plugin_update (FuPlugin *plugin,
 	g_autoptr(GError) error_local = NULL;
 	gboolean force = (flags & FWUPD_INSTALL_FLAG_FORCE) != 0;
 	FuPluginValidation validation;
+	gboolean force_powered = FALSE;
 
 	devpath = fu_device_get_metadata (dev, "sysfs-path");
 	g_return_val_if_fail (devpath, FALSE);
 
 	udevice = g_udev_client_query_by_sysfs_path (data->udev, devpath);
 	if (udevice == NULL) {
-		g_set_error (error,
-			     FWUPD_ERROR,
-			     FWUPD_ERROR_NOT_FOUND,
-			     "could not find thunderbolt device at %s",
-			     devpath);
-		return FALSE;
+		if (fu_plugin_thunderbolt_can_force_power (plugin)) {
+			fu_plugin_thunderbolt_set_force_power (plugin, TRUE, error);
+			force_powered = TRUE;
+			g_usleep (FU_PLUGIN_THUNDERBOLT_UPDATE_TIMEOUT_MS);
+			udevice = g_udev_client_query_by_sysfs_path (data->udev, devpath);
+		}
+
+		if (udevice == NULL) {
+			if (force_powered)
+				fu_plugin_thunderbolt_set_force_power (plugin, FALSE, error);
```

Comment:
Check for error? Especially as it can fail the next set error.

Reply:
Checked in new plugin